VirtoCommerce.MarketingModule.Web/Scripts/promotion/blades/catalog-entry-list.js

```diff
@@ -0,0 +1,139 @@
+angular.module('virtoCommerce.marketingModule')
+    .controller('virtoCommerce.marketingModule.catalogEntriesController', ['$scope', 'virtoCommerce.catalogModule.items', 'platformWebApp.ui-grid.extension', 'platformWebApp.bladeUtils', 'platformWebApp.uiGridHelper', 'platformWebApp.bladeNavigationService',
+        function ($scope, items, gridOptionExtension, bladeUtils, uiGridHelper, bladeNavigationService) {
+            var blade = $scope.blade;
+
+            $scope.isEntityChanged = function () {
+                return angular.equals(this.blade.productIds, this.blade.currentProductIds);
+            };
+
+            $scope.saveChanges = function () {
+                this.blade.promotion.productIds = this.blade.productIds.slice();
+                bladeNavigationService.closeBlade(this.blade);
+            };
+
+            $scope.setGridOptions = function (gridId, gridOptions) {
+                gridOptionExtension.tryExtendGridOptions(gridId, gridOptions);
+                initialize();
+                uiGridHelper.initialize($scope, gridOptions, externalRegisterApiCallback);
+                bladeUtils.initializePagination($scope);
+            };
+
+            blade.refresh = function () {
+                initialize();
+            };
+
+            $scope.cancel = function () {
+                initialize();
+            };
+
+            function initialize() {
+                blade.productIds = blade.promotion.productIds != undefined ? blade.promotion.productIds.slice() : [];
```

Comment:
"blade.promotion.productIds != undefined" could be simplified with "blade.promotion.productIds"